### PR TITLE
[Nala]: Fix split view separator colors

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/40518

|  | Before | After |
|--------|--------|--------|
| Light | Screenshot from 2024-08-19 11-34-35 | Screenshot from 2024-08-19 11-26-28 |
| Dark | image | Screenshot from 2024-08-19 11-26-44 |
| Private | image | Screenshot from 2024-08-19 11-32-25 |
| Tor | image | Screenshot from 2024-08-19 11-32-37 | 

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

See issue

### CODE REVIEW

File: ui/color/missing_color_ref_mixer.cc
Comment (fallaciousreasoning) [inline ui/color/missing_color_ref_mixer.cc:1]: Changes to this file are semi-unrelated - I can split them into a separate PR if that's easier: Basically, we weren't setting these colors in the dynamic palette.

File: browser/ui/views/split_view/split_view_separator.cc [hunk 1]
```diff
@@ -184,6 +184,14 @@ void SplitViewSeparator::OnBoundsChanged(const gfx::Rect& previous_bounds) {
   LayoutMenuButton();
 }
 
+void SplitViewSeparator::OnPaintBackground(gfx::Canvas* canvas) {
+  auto* cp = GetColorProvider();
```
Comment (fallaciousreasoning) [inline browser/ui/views/split_view/split_view_separator.cc:188]: Not really sure why this was working before - we didn't seem to be setting the background

File: browser/ui/tabs/brave_tab_color_mixer.cc [hunk 2]
```diff
@@ -97,23 +120,8 @@ void AddBraveTabLightThemeColorMixer(ui::ColorProvider* provider,
           {kColorBraveVerticalTabNTBShortcutTextColor,
            SkColorSetRGB(0x85, 0x89, 0x89)},
           {kColorBraveSplitViewTileBackground, SkColorSetRGB(0xDA, 0xDF, 0xE1)},
-          {kColorBraveSharedPinnedTabDummyViewBackground,
```
Comment (fallaciousreasoning) [inline browser/ui/tabs/brave_tab_color_mixer.cc:100]: Here, I had mistakenly passed in a ColorId where it was expecting a SkColor

File: browser/ui/views/split_view/split_view_separator.cc (hunk 1 above)
Comment (simonhong) [reply browser/ui/views/split_view/split_view_separator.cc:189]: I assume that separator's bg color comes frome its parent?

**Review by simonhong — APPROVED** [review]

File: browser/ui/views/split_view/split_view_separator.cc (hunk 1 above)
Comment (fallaciousreasoning) [reply browser/ui/views/split_view/split_view_separator.cc:189]: It's parent was a widget, and it was see through before this change. Maybe that bit came from upstream?

Comment (fallaciousreasoning) [reply browser/ui/views/split_view/split_view_separator.cc:189]: (it was fine in Beta, not in Nightly, but that could be related to Nala)

Comment (simonhong) [reply browser/ui/views/split_view/split_view_separator.cc:189]: `kColorBraveSplitViewMenuButtonBackground` is that color. it fills that that widget's bg.

Comment (simonhong) [reply browser/ui/views/split_view/split_view_separator.cc:189]: oh, wait, this seems not related with menu button?

Comment (fallaciousreasoning) [reply browser/ui/views/split_view/split_view_separator.cc:189]: Yeah, it's the divider line, not the button :smile:

**Pull request merged**